remoulade/state/backends/redis.py

```diff
@@ -26,25 +26,42 @@ def __init__(self, *, namespace="remoulade-state", encoder=None, client=None, ur
         super().__init__(namespace=namespace, encoder=encoder)
         if url:
             parameters["connection_pool"] = redis.ConnectionPool.from_url(url)
+        parameters["decode_responses"] = True
         self.client = client or redis.Redis(**parameters)
 
     def get_state(self, message_id):
         key = self._build_message_key(message_id)
-        data = self.client.get(key)
-        if data:
+        data = self.client.hgetall(key)
+        if len(data) == 0:
+            data = None
+        else:
             data = self._parse_state(data)
         return data
 
     def set_state(self, state, ttl):
         message_key = self._build_message_key(state.message_id)
-        self.client.set(message_key, self.encoder.encode(state.asdict()), ex=ttl)
+        with self.client.pipeline() as pipe:
```

Comment:
I would prefer i there was only the method set_state.
Then state.asdict() could return a dict with only the keys which have a value different of None.
This way you can create and update the state with the same function.